databroker/experimental/server_ext.py

```diff
@@ -62,8 +63,9 @@ def inner(self, *args, **kwargs):
 class WritingArrayAdapter:
     structure_family = "array"
 
-    def __init__(self, collection, directory, doc):
+    def __init__(self, collection, revisions, directory, doc):
```

Comment:
I think it would make sense to pass in the `database` here rather than separately passing in each of its collections.

Reply:
Fixed in the latest commit